core/src/test/java/com/ericsson/bss/cassandra/ecchronos/core/metrics/TestMetricInspector.java

```diff
@@ -0,0 +1,125 @@
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.read.ListAppender;
+import com.ericsson.bss.cassandra.ecchronos.core.TableStorageStates;
+import com.ericsson.bss.cassandra.ecchronos.core.utils.StatusLogger;
+import com.ericsson.bss.cassandra.ecchronos.core.utils.TableReference;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.slf4j.LoggerFactory;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import static com.ericsson.bss.cassandra.ecchronos.core.MockTableReferenceFactory.tableReference;
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestMetricInspector
+{
+    private static final String TEST_KEYSPACE = "test_keyspace";
+    private static final String TEST_TABLE1 = "test_table1";
+    private static final String TEST_TABLE2 = "test_table2";
+    @Mock
+    private TableStorageStates myTableStorageStates;
+    private MeterRegistry myMeterRegistry;
+    private TableRepairMetricsImpl myTableRepairMetricsImpl;
+    private MetricInspector myMetricInspector;
+    private LoggerContext loggerContext;
+    private ListAppender<ILoggingEvent> listAppender;
+
+    @Before
+    public void init()
+    {
+        loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+        listAppender = new ListAppender<>();
+        listAppender.start();
+        loggerContext.getLogger(StatusLogger.class).addAppender(listAppender);
+        // Use composite registry here to simulate real world scenario where we have multiple registries
+        CompositeMeterRegistry compositeMeterRegistry = new CompositeMeterRegistry();
+        // Need at least one registry present in composite to record metrics
+        compositeMeterRegistry.add(new SimpleMeterRegistry());
+        myMeterRegistry = compositeMeterRegistry;
+        myTableRepairMetricsImpl = TableRepairMetricsImpl.builder()
+                .withTableStorageStates(myTableStorageStates)
+                .withMeterRegistry(myMeterRegistry)
+                .build();
+        myMetricInspector = new MetricInspector(myMeterRegistry,
+                1, 1,5000);
+    }
+
+    @Test
+    public void testInspectMeterRegistryForRepairFailuresWhenFailureThresholdIsBroken()
+    {
+        TableReference tableReference = tableReference(TEST_KEYSPACE, TEST_TABLE1);
+        long expectedRepairTime = 12345L;
+        myTableRepairMetricsImpl.repairSession(tableReference, expectedRepairTime, TimeUnit.MILLISECONDS, false);
+        TableReference tableReference1 = tableReference(TEST_KEYSPACE, TEST_TABLE1);
```

Comment:
You're not using "tableReference1" in this test? Why?

It is confusing when you name some variables without a number and some with and they're essentially referencing to the same table (TEST_TABLE1). Eg:
tableReference
tableReference1

Why not?
tableReference1
tableReference2